### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by wtritch**
Just some minor fixes I found while familiarizing myself with the code base

### CODE REVIEW

File: webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java [hunk 1]
```diff
@@ -50,18 +54,29 @@ public StudentGroupBatch upload(final User user, final MultipartFile file) {
 
         final StudentGroupBatch.Builder studentGroupBatchBuilder = StudentGroupBatch.builder().creator(user.getUsername());
 
-        try {
-            final String digest = DigestUtils.md5Hex(validate(file.getBytes())).toUpperCase();
+        final MessageDigest messageDigest = DigestUtils.getMd5Digest();
+        try (final DigestInputStream digestStream = new DigestInputStream(file.getInputStream(), messageDigest)) {
+            validate(digestStream);
```
Comment (wtritch) [inline webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java:59]: Changed this to use a DigestInputStream and validate the CSV against an InputStream as well, so we never have to load the payload fully into memory.

File: webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java [hunk 2]
```diff
@@ -50,18 +54,29 @@ public StudentGroupBatch upload(final User user, final MultipartFile file) {
...
+                    .status(ImportStatus.BAD_DATA);
+            return repository.create(studentGroupBatchBuilder.build());
+        }
+
+        final String digest = Hex.encodeHexString(messageDigest.digest()).toUpperCase();
+        try (final InputStream fileStream = file.getInputStream()) {
             final Properties properties = new Properties();
-            properties.setProperty("content-type", file.getContentType());
+            properties.put(Headers.CONTENT_TYPE, file.getContentType());
+            properties.put(Headers.CONTENT_LENGTH, file.getSize());
```
Comment (wtritch) [inline webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java:72]: The ArchiveService expects a Long value for the CONTENT_LENGTH.  I just changed this to use the well-known Amazon header keys and set the CONTENT_LENGTH as a Long value.

File: webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java [hunk 3]
```diff
@@ -50,18 +54,29 @@ public StudentGroupBatch upload(final User user, final MultipartFile file) {
...
             properties.setProperty("username", user.getUsername());
-            properties.setProperty("size", Long.toString(file.getSize()));
             properties.setProperty("filename", file.getName());
 
             //TODO: check if the digest already exists
 
             final String location = new LocationStrategy.GroupUploadContentLocationStrategy().location(digest);
-            archiveService.writeResource(location, file.getInputStream(), properties);
```
Comment (wtritch) [inline webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java:64]: We weren't closing the InputStream before.

File: webapp/src/test/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchServiceTest.java [hunk 4]
```diff
@@ -13,14 +14,32 @@
...
+            "subject_code",
+            "group_name",
+            "group_user_login",
+            "school_year",
+            "student_ssid"
+    );
+
+    private static final List<String> ValidValues = ImmutableList.copyOf(ValidHeaders.stream()
+            .map(header -> header + "_val")
+            .collect(Collectors.toList()));
```
Comment (wtritch) [inline webapp/src/test/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchServiceTest.java:42]: I was starting to get lost in the super-long embedded CSV strings.  Minor re-working to use lists and Joiners.

File: webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java [hunk 5]
```diff
@@ -1,5 +1,7 @@
 package org.opentestsystem.rdw.admin.service.impl;
 
+import com.amazonaws.services.s3.Headers;
```
Comment (malaffoon) [inline webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java:3]: dragging amazonaws in just for common http headers seems wrong

File: webapp/src/test/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchServiceTest.java [hunk 6]
```diff
@@ -56,32 +75,37 @@ public void itShouldUseRepositoryToFindAllStudentGroupBatches() throws IOExcepti
...
     public void itShouldUpload() throws IOException {
-        when(file.getBytes()).thenReturn(("school_natural_id,subject_code,group_name,group_user_login,school_year,student_ssid\nschool_natural_id_val,subject_code_val,group_name_val,group_user_login_val,school_year_val,student_ssid_val\n").getBytes());
+        final String payload = payload(ValidHeaders, ValidValues);
+        when(file.getInputStream()).thenReturn(new ByteArrayInputStream(payload.getBytes()));
         final StudentGroupBatch batch = StudentGroupBatch.builder()
                 .creator(user.getUsername())
                 .status(ImportStatus.ACCEPTED)
-                .digest(DigestUtils.md5Hex(file.getBytes())
+                .digest(DigestUtils.md5Hex(payload.getBytes())
                         .toUpperCase())
```
Comment (wtritch) [inline webapp/src/test/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchServiceTest.java:84]: This proves that the DigestInputStream produces the same digest as calling it against a byte array.

**Review by malaffoon — APPROVED** [review]

File: webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java (hunk 5 above)
Comment (wtritch) [reply webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java:3]: The LocalArchiveService and S3ArchiveService are both using that value to key off of... it seemed reasonable to me.  Objections?

Comment (malaffoon) [reply webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java:3]: No.

**Pull request merged**